cpp/CMakeLists.txt

```diff
@@ -334,6 +367,7 @@ add_library(cugraph SHARED
 #        wait until after cugunrock is constructed.
 #
 add_dependencies(cugraph cugunrock)
+add_dependencies(cugraph raft)
 
 if (BUILD_MPI)
```

Comment:
`if (BUILD_MPI)` should not impact RAFT.

Recall that this is false on CI so CI will ignore this RAFT integration if we leave this as is. We should : 
- Disable MPI code (or keep it encapsulated)
- Enable NCCL and RAFT by default and test through CI.